libs/community/tests/integration_tests/vectorstores/test_falkordbvector.py

```diff
@@ -0,0 +1,596 @@
+        host=host,
+        port=port,
+        node_label="Test",
+        text_node_properties=["name", "name2"],
+        embedding_node_property="embedding",
+        search_type=SearchType.HYBRID,
+    )
+
+    output = existing.similarity_search("foo", k=2)
+
+    assert [output[0]] == [Document(page_content='\nname: Foo\nname2: Fooz')]
+
+    drop_vector_indexes(existing)
+
+
+def test_index_fetching() -> None:
+    """testing correct index creation and fetching"""
+    embeddings = FakeEmbeddings()
+    
+    def create_store(
+        node_label: str, text_properties: List[str]
+    ) -> FalkorDBVector:
+        return FalkorDBVector.from_existing_graph(
+            embedding=embeddings,
+            host=host,
+            port=port,
+            node_label=node_label,
+            text_node_properties=text_properties,
+            embedding_node_property="embedding",
+        )
+    
+    def fetch_store(node_label: str) -> FalkorDBVector:
+        store = FalkorDBVector.from_existing_index(
+            embedding=embeddings,
+            host=host,
+            port=port,
+            node_label=node_label,
+        )
+        return store
+    
+    index_0_str = "label0"
+    create_store(index_0_str, ["text"])
+
+    # create index 1
+    index_1_str = "label1"
+    create_store("label1", ["text"])
+
+    index_1_store = fetch_store(index_1_str)
+    assert index_1_store.node_label == index_1_str
+
+    index_0_store = fetch_store(index_0_str)
+    assert index_0_store.node_label == index_0_str
+
+    drop_vector_indexes(index_1_store)
+    drop_vector_indexes(index_0_store)
+
+
+def test_retrieval_params() -> None:
+    """Test if we use parameters in retrieval query"""
+    docsearch = FalkorDBVector.from_texts(
+        texts=texts,
+        embedding=FakeEmbeddings(),
+        pre_delete_collection=True,
+        retrieval_query="""
+        RETURN $test as text, score, {test: $test1} AS metadata
+        """,
+    )
+    output = docsearch.similarity_search(
+        "Foo", k=2, params={"test": "test", "test1": "test1"}
+    )
+    assert output == [
+        Document(page_content="test", metadata={"test": "test1"}),
+        Document(page_content="test", metadata={"test": "test1"}),
+    ]
+    drop_vector_indexes(docsearch)
+
+
+
+def test_falkordb_relationship_index() -> None:
+    """Test end to end construction and search."""
+    embeddings = FakeEmbeddingsWithOsDimension()
+    docsearch = FalkorDBVector.from_texts(
+        texts=texts,
+        embedding=embeddings,
+        host=host,
+        port=port,
+        pre_delete_collection=True,
+    )
+    # Ingest data
+    docsearch.query(
+        (
+            "MERGE (p1:Person)"
+            "MERGE (p2:Person)"
+            "MERGE (p3:Person)"
+            "MERGE (p4:Person)"
+            "MERGE (p1)-[:REL {text: 'foo', embedding: vecf32($e1)}]->(p2)"
+            "MERGE (p3)-[:REL {text: 'far', embedding: vecf32($e2)}]->(p4)"
```

Comment:
I see no reason to...If the graph is empty MERGE with create the nodes and relationships with its ON CREATE functionality else it would match the nodes with its ON MATCH functionality